### PR TITLE
Add prop renderEmptyItems to the List component

### PR DESCRIPTION
**Pull request by yurist38**
# What does this PR do:

    Added prop `renderEmptyItems` to the `List` component. Value by default is `true`.

# Where should the reviewer start:

    Diff

### CODE REVIEW

File: components/list/list.js [hunk 1]
```diff
@@ -32,33 +42,39 @@ function List(props) {
 }
 
 List.defaultProps = {
-  hideBullets: false,
   align: 'vertical',
+  hideBullets: false,
+  renderEmptyItems: true,
```
Comment (asci) [inline components/list/list.js]: any case when it should be `true` by default?

Comment (yurist38) [reply components/list/list.js]: Not sure that there is any. I did that way just to save existing functionality.

File: components/list/list.js [hunk 2]
```diff
@@ -18,7 +24,11 @@ function List(props) {
 
   const className = getClassNamesWithMods('ui-list', mods);
 
-  const itemsBlock = items.map((item, index) => (
+  const notEmpty = i => i !== '' && i !== null;
+
+  const itemsToRender = (renderEmptyItems === false) ? items.filter(notEmpty) : items;
```
Comment (mAiNiNfEcTiOn) [inline components/list/list.js]: Good one. You could've also filter by `Boolean`:

```js
items.filter(Boolean)
```

Comment (yurist38) [reply components/list/list.js]: But it that case we will get 0 values filtered out, I wanted to prevent that. @mAiNiNfEcTiOn

**Pull request merged**